docs/technical-reference/tasks.md

```diff
@@ -46,14 +46,41 @@ For the API reference visit [TextGenerationTask][distilabel.tasks.text_generatio
 The task specially designed to build the prompts following the Self-Instruct paper: [SELF-INSTRUCT: Aligning Language Models
 with Self-Generated Instructions](https://arxiv.org/pdf/2212.10560.pdf).
 
-From the original [repository](https://github.com/yizhongw/self-instruct/tree/main#how-self-instruct-works): *The Self-Instruct process is an iterative bootstrapping algorithm that starts with a seed set of manually-written instructions and uses them to prompt the language model to generate new instructions and corresponding input-output instances*, so this `Task` is specially interesting for generating new datasets from a set of predefined topics.
+From the original [repository](https://github.com/yizhongw/self-instruct/tree/main#how-self-instruct-works):
+
+*The Self-Instruct process is an iterative bootstrapping algorithm that starts with a seed set of manually-written instructions and uses them to prompt the language model to generate new instructions and corresponding input-output instances*, so this `Task` is specially interesting for generating new datasets from a set of predefined topics.
 
 ```python
 --8<-- "docs/snippets/technical-reference/tasks/generic_openai_self_instruct.py"
 ```
 
 For the API reference visit  [SelfInstructTask][distilabel.tasks.text_generation.self_instruct.SelfInstructTask].
 
+### EvolInstructTask
+
+The task specially designed to build the prompts following the Evol-Instruct strategy proposed in: [WizardLM: Empowering Large Language Models to
+Follow Complex Instructions](https://arxiv.org/abs/2304.12244).
+
+From the original [repository](https://github.com/nlpxucan/WizardLM/tree/main?tab=readme-ov-file#overview-of-evol-instruct): 
+
+*Evol-Instruct is a novel method using LLMs instead of humans to automatically mass-produce open-domain instructions of various difficulty levels and skills range, to improve the performance of LLMs*.
+
+Use this `Task` to build more complete and complex datasets starting from simple ones.
+
+```python
+--8<-- "docs/snippets/technical-reference/tasks/generic_openai_evol_instruct.py"
+```
+
+You can take a look at a [sample dataset](https://huggingface.co/datasets/argilla/distilabel-sample-evol-instruct?row=19) generated using the script the following script: [examples/pipeline-evol-instruct-alpaca.py](../../examples/pipeline-evol-instruct-alpaca.py).
+
+!!! note
+    The original definition of `EvolInstruct` considers an elimination evolving step with different
+    situations to remove the responses considered as failures. Section 3.2, *Elimination Evolving* in [WizardLM paper](https://arxiv.org/abs/2304.12244) shows these steps. We have implemented steps 2-4 as part of this task, but not step one. If the user wants to tackle this, an example can be seen in the following script: [examples/pipeline-openai-wizardl-equal-prompts.py](../../examples/pipeline-evol-instruct-alpaca.py).
```

Comment:
the wrong file is referenced here

Reply:
also mention more explicitly that it shows how to implement evaluation step 1 using distilabel please